common/app/templates/inlineJS/blocking/enableStylesheets.scala.js

```diff
@@ -1,38 +1,126 @@
+        // `media` [OPTIONAL] is the media type or query of the stylesheet. By default it will be 'all'
+        var doc = w.document;
+        var ss = doc.createElement( 'link' );
+        var ref;
+
+        if (before) {
+            ref = before;
+        } else {
+            var refs = (doc.body || doc.getElementsByTagName( 'head' )[ 0 ]).childNodes;
+            ref = refs[refs.length - 1];
+        }
+
+        var sheets = doc.styleSheets;
+        ss.rel = 'stylesheet';
+        ss.href = href;
+        // temporarily set media to something inapplicable to ensure it'll fetch without blocking render
+        ss.media = 'only x';
+
+        // wait until body is defined before injecting link. This ensures a non-blocking load in IE11.
+        function ready( cb ) {
+            if (doc.body) {
+                return cb();
             }
+
+            setTimeout(function() {
+                ready(cb);
+            });
         }
-        requestAnimationFrame(function () {
-            setMedia(styleSheet);
+
+        // Inject link
+        // Note: the ternary preserves the existing behavior of 'before' argument, but we could choose to change the argument to 'after' in a later release and standardize on ref.nextSibling for all refs
+        // Note: `insertBefore` is used instead of `appendChild`, for safety re: http://www.paulirish.com/2011/surefire-dom-element-insertion/
+        ready(function() {
+            ref.parentNode.insertBefore(ss, (before ? ref : ref.nextSibling));
         });
 
-    }
+        // A method (exposed on return object for external use) that mimics onload by polling document.styleSheets until it includes the new sheet.
+        var onloadcssdefined = function(cb) {
+            var resolvedHref = ss.href;
+            var i = sheets.length;
+            while (i--) {
+                if (sheets[ i ].href === resolvedHref){
+                    return cb();
+                }
+            }
+
+            setTimeout(function() {
+                onloadcssdefined( cb );
+            });
+        };
 
-    // Watch for load on all `link` elements with media of `only x`
-    function useCss () {
-        for (var i = 0, totalStyleSheetLinks = styleSheetLinks.length; i < totalStyleSheetLinks ; i++) {
-            if (styleSheetLinks[i].getAttribute('media') === 'only x') {
-                setMedia(styleSheetLinks[i]);
+        function loadCB() {
+            if (ss.addEventListener) {
+                ss.removeEventListener('load', loadCB);
             }
+
+            ss.media = media || 'all';
+            exposeLoadStatus();
+        }
+
+        // once loaded, set link's media back to `all` so that the stylesheet applies once it loads
+        if (ss.addEventListener) {
+            ss.addEventListener('load', loadCB);
         }
+
+        ss.onloadcssdefined = onloadcssdefined;
+        onloadcssdefined(loadCB);
+
+        return ss;
+    };
+
+    preloadSpported = function() {
```

Comment:
Will the implicit `var` attach this to the `window`? Also is the typo intentional? 😉